src/Type/Definition/PhpEnumType.php

```diff
@@ -39,22 +43,29 @@ public function __construct(string $enum, ?string $name = null)
         }
 
         parent::__construct([
-            'name' => $name ?? $this->baseName($enum),
+            'name' => $config['name'] ?? $this->baseName($config['enumPath']),
             'values' => $enumDefinitions,
-            'description' => $this->extractDescription($reflection),
+            'description' => $config['description'] ?? $this->extractDescription($reflection),
+            'enumPath' => $config['enumPath']
         ]);
     }
 
     public function serialize($value): string
     {
-        if (! is_a($value, $this->enumClass)) {
+        if (! is_a($value, $this->config['enumPath'])) {
             $notEnum = Utils::printSafe($value);
-            throw new SerializationError("Cannot serialize value as enum: {$notEnum}, expected instance of {$this->enumClass}.");
+            throw new SerializationError("Cannot serialize value as enum: {$notEnum}, expected instance of {$this->config['enumPath']}.");
         }
-
         return $value->name;
     }
 
+    /**
+    /* @param class-string $path
+     */
+    public static function fromPath($path): PhpEnumType {
+        return new static(["enumPath"=>$path]);
+    }
```

Comment:
I don't think this is necessary

Reply:
Heh, yeah I wasn't crazy about it, either. 👍